Core/Core/Views/GradeCircleView.swift

```diff
@@ -0,0 +1,128 @@
+//
+// Copyright (C) 2019-present Instructure, Inc.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, version 3 of the License.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+//
+
+import UIKit
+
+public class GradeCircleView: UIView {
+    @IBOutlet weak var circlePoints: UILabel!
+    @IBOutlet weak var circleLabel: UILabel!
+    @IBOutlet weak var circleComplete: UIImageView!
+    @IBOutlet weak var gradeCircle: GradeCircle?
+    @IBOutlet weak var displayGrade: UILabel!
+    @IBOutlet weak var outOfLabel: UILabel!
+    @IBOutlet weak var latePenaltyLabel: UILabel!
+    @IBOutlet weak var finalGradeLabel: UILabel!
+
+    public override init(frame: CGRect) {
+        super.init(frame: frame)
+        loadFromXib()
+    }
+
+    public required init?(coder aDecoder: NSCoder) {
+        super.init(coder: aDecoder)
+        loadFromXib()
+    }
+
+    public func update(_ assignment: Assignment) {
+        // in this case the submission should always be there because canvas generates
+        // submissions for every user for every assignment but just in case
+        guard let submission = assignment.submission, submission.workflowState != .unsubmitted else {
+            self.isHidden = true
+            return
+        }
+
+        guard submission.grade != nil else {
+            self.isHidden = true
+            return
+        }
+
+        self.isHidden = false
```

Comment:
This is a knitpick.  Using `self` in swift has an older `objective-c` feel to it.  Sometimes using `self` is required like in an `init` but most of the time it's not needed.  I think most(?) of swift our code generally follows this style of omitting self when not needed.

Reply:
👍 I'm not opposed to fixing this.

Reply:
Up to you if you want to fix or not.  I'm fine approving if not.